src/backend/distributed/citus--7.3-2--7.3-3.sql

```diff
@@ -0,0 +1,75 @@
+/* citus--7.3-2--7.3-3 */
+
+/*****************************************************************************
+ * Citus json aggregate helpers
+ *****************************************************************************/
+
+CREATE FUNCTION pg_catalog.citus_jsonb_concatenate(state jsonb, val jsonb)
+	RETURNS jsonb
+	LANGUAGE SQL
+AS $function$
+	SELECT CASE 
+		WHEN val IS NULL THEN state
+		WHEN jsonb_typeof(state) = 'null' THEN val
```

Comment:
Did you run into a specific problem that made you treat a JSONB `null` differently? Or do you think it's more conceptually sound?

Reply:
Postgres `array_cat` and `jsonb_concat` treat `NULL` concatenation differently. `jsonb_concat` nullifies the value if we try to append `NULL` into to the object/array. Therefore we need to handle these cases some way or another.

Reply:
Yep, `jsonb_concat` is unfortunately marked as `STRICT` which means that it always returns `NULL` if an argument is `NULL`, whereas `array_cat` is not. Defining a wrapper that handles NULL input is the right thing to do.

though jsonb_concat does treat 'null'::jsonb as a regular value
```sql
postgres=# SELECT jsonb_concat('null'::jsonb, '3'::jsonb);
 jsonb_concat 
──────────────
 [null, 3]
(1 row)

Time: 0.293 ms
postgres=# SELECT citus_jsonb_concatenate('null'::jsonb, '3'::jsonb);
 citus_jsonb_concatenate 
─────────────────────────
 3
(1 row)
```
so that made me wonder if there's anything special about `'null'::jsonb` in the Citus case. I realise that `'null'::jsonb` is the `INITCOND` of the aggregate so it's used to bootstrap the aggregate. On the other hand, you could have also used `NULL` as the `INITCOND` and check for `state IS NULL` here.

I think either way is fine, since the difference is philosphical and it perfectly solves the problem of distributing `jsonb_agg`, but was just making sure that I'm not missing anything.

Reply:
There is something intresting going on when using `NULL` as the `INITCOND` here. It actually gets converted into `'null'::jsonb` somewhere along the way. So checking `state IS NULL` does not work. I really can not explain what happens in this case behind the scenes. So that PG aggregate initialization gets `NULL` actually converted to `'null'::jsonb`. I then chose to be explicit about the initial condition and set it to be `'null'`.

Reply:
Because of this the `'null'::jsonb` is used to detect whether anything was received via aggregate state transition. Then final function yields actual `NULL` if nothing was received (state is `'null'::jsonb`).